### PR TITLE
Add appsettings & environment variables

### PR DESCRIPTION
**Pull request by grygorenkod**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: Streetcode/Streetcode/Program.cs [hunk 1]
```diff
@@ -1,14 +1,30 @@
...
 using Services.Interfaces;
 using Services.Services;
+using System;
+
+// for Nuke
+//environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
+//if (environment != "Local")
+//{
+//    Environment.SetEnvironmentVariable("ASPNETCORE_ENVIRONMENT", "Local", EnvironmentVariableTarget.Machine);
+//}
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode/Program.cs:13]: You can remove these comments, these should be in Nuke SetupEnvironmentVariables target

File: Streetcode/Streetcode/appsettings.Development.json [hunk 2]
```diff
@@ -1,8 +1,6 @@
 {
-  "Logging": {
-    "LogLevel": {
-      "Default": "Information",
-      "Microsoft.AspNetCore": "Warning"
-    }
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode/appsettings.Development.json:6]: Let's move these logging overriding to appsettings.json

File: Streetcode/Streetcode/appsettings.IntegrationTests.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "ConnectionStrings": {
+    "DefaultConnection": "Server=(localdb)\\MSSQLLocalDB;Database=_CHANGE_ME;Trusted_Connection=True;MultipleActiveResultSets=true"
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode/appsettings.IntegrationTests.json:3]: Let's rename this "DefaultConnection" to "Main" and in future you will have to add here connection string to your database in docker container

File: Streetcode/Streetcode/appsettings.Production.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "URLs": {
+    "BackEnd": "BackUrl",
+    "FrontEnd": "FrontUrl"
+  }
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode/appsettings.Production.json:5]: You can remove Backend url

File: Streetcode/Streetcode/Program.cs [hunk 5]
```diff
@@ -19,16 +35,19 @@
 // Configure the HTTP request pipeline.
 if (app.Environment.IsDevelopment())
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode/Program.cs:36]: Let's use that only in Local env

File: Streetcode/Streetcode/Program.cs [hunk 6]
```diff
@@ -19,16 +35,19 @@
...
 {
+    builder.Configuration.AddUserSecrets<string>();
     app.UseSwagger();
     app.UseSwaggerUI();
+    
+}
+if (!app.Environment.IsDevelopment())
+{
+    app.UseExceptionHandler("/Home/Error");
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode/Program.cs:45]: Do we have this error handling page?

File: Streetcode/Streetcode/Program.cs [hunk 7]
```diff
@@ -19,16 +35,19 @@
 // Configure the HTTP request pipeline.
 if (app.Environment.IsDevelopment())
 {
+    builder.Configuration.AddUserSecrets<string>();
     app.UseSwagger();
     app.UseSwaggerUI();
+    
+}
+if (!app.Environment.IsDevelopment())
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode/Program.cs:43]: I guess here should be Local instead of Development

**Review by LanchevychMaxym — CHANGES_REQUESTED** [review]

File: Streetcode/Streetcode/Program.cs [hunk 8]
```diff
@@ -1,34 +1,41 @@
...
 var builder = WebApplication.CreateBuilder(args);
+var environment = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT");
 
-// Add services to the container.
+builder.Host.ConfigureAppConfiguration((hostingContext, config) =>
+{
+    config.SetBasePath(Directory.GetCurrentDirectory());
+    config.AddJsonFile("appsettings.json", optional: false, reloadOnChange: true);
+    config.AddJsonFile($"appsettings.{environment}.json", optional: true, reloadOnChange: true);
+    config.AddEnvironmentVariables("STREETCODE_");
```
Comment (dimasster) [inline Streetcode/Streetcode/Program.cs:12]: why we need _ in the end of variable, maybe it is malformed name

File: Streetcode/Streetcode/Properties/launchSettings.json [hunk 9]
```diff
@@ -1,26 +1,42 @@
...
+        "STREETCODE_ENVIRONMENT": "IntegrationTests"
+      }
+    },
+    "Streetcode_Prod": {
+      "commandName": "Project",
       "launchBrowser": true,
-      "launchUrl": "swagger",
       "environmentVariables": {
-        "ASPNETCORE_ENVIRONMENT": "Development"
+        "ASPNETCORE_ENVIRONMENT": "Production"
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode/Properties/launchSettings.json:34]: This should be local

File: Streetcode/Streetcode/appsettings.Local.json [hunk 10]
```diff
@@ -0,0 +1,5 @@
+{
+  "ConnectionStrings": {
+    "DefaultConnection": "Server=localhost;Database=StreetcodeDb;UserId=sa;Password=Admin@1234;MultipleActiveResultSets=true"
```
Comment (LanchevychMaxym) [inline Streetcode/Streetcode/appsettings.Local.json:3]: In all configs this section should have the same name "Main"

**Review by LanchevychMaxym — APPROVED** [review]

**Review by Tatiana2424 — APPROVED** [review]

**Review by dimasster — APPROVED** [review]

**Review by AleXLaeR — APPROVED** [review]

**Pull request merged**